src/pyvesync/vesync.py

```diff
@@ -93,23 +115,66 @@ def get_energy_dict_from_api(self, response):
         return data
 
     def get_headers(self):
-        return {'Content-Type': 'application/json'}
+        header = {}
+        header['accept-language'] = 'en'
+        header['accountId'] = self.account_id
+        header['appVersion'] = APP_VERSION
+        header['content-type'] = 'application/json'
+        header['tk'] = self.tk
+        header['tz'] = self.time_zone
+        return header
+
+    def get_body(self, type):
+        bodybase = {}
+        bodybase['timeZone'] = self.time_zone
+        bodybase['acceptLanguage'] = 'en'
+
+        bodyauth = {}
+        bodyauth['accountID'] = self.account_id
+        bodyauth['token'] = self.tk
+
+        bodydetails = {}
+        bodydetails['appVersion'] = APP_VERSION
+        bodydetails['phoneBrand'] = PHONE_BRAND
+        bodydetails['phoneOS'] = PHONE_OS
+        bodydetails['traceId'] = str(int(time.time()))
+
+        if type == 'login':
+            login = dict(**bodybase, **bodydetails)
+            login['devToken'] = ''
+            login['userType'] = USER_TYPE
+            login['method'] = 'login'
+            return login
+        elif type == 'devicelist':
+            devlist = dict(**bodybase, **bodyauth, **bodydetails)
+            devlist['method'] = 'devices'
+            devlist['pageNo'] = '1'
+            devlist['pageSize'] ='50'
+            return devlist
+        elif type ==  'devicedetail':
+            devdetail = dict(**bodybase, **bodyauth, **bodydetails)
+            return devdetail
+        elif type == 'devicestatus':
+            devstatus = dict(**bodybase, **bodyauth)
+            return devstatus
 
     def login(self):
         """Return True if log in request succeeds"""
 
         try:
-            jd = {'account': self.username, 'password': hashlib.md5(self.password.encode('utf-8')).hexdigest()}
+            jd = {'email': self.username, 'password': hashlib.md5(self.password.encode('utf-8')).hexdigest()}
+            body = self.get_body('login')
+            body.update(jd)
         except ValueError:
             logger.error("Unable to read username and password")
 
             return False
         else:
-            response, _ = self.call_api('/vold/user/login', 'post', json=jd)
+            response, _ = self.call_api('/cloud/v1/user/login', 'post', json=body)
 
-            if response is not None and response and 'tk' in response and 'accountID' in response:
-                self.tk = response['tk']
-                self.account_id = response['accountID']
+            if response and response['code'] == 0:
```

Comment:
might be risky to rely on this code response item being '0' without knowing the details of their api strategy. is it a placeholder that they are setting to 0 right now as something they will change later or does 0 have actual meaning right now?
it might be safer to instead check that the response is a 200 and then return true if the response contains a token and accountID value but as it stands now i think what you have is still fine

Reply:
From what I've seen how the api works the response code will always be 200 even if there is an error (except with 7A switches, which I believe they are moving away from those types of calls).  The code will always be greater than 0 no matter what the error is.  I've started with the VeSync subclass check_response.  The goal is to centralize the function that checks responses for all the calls, so that if they ever do a sweeping change, we can fix that one function instead of having to check each call within the switch types.
I'm working on making this more robust right now, so I'll have a commit in shortly.  Right now the code 0 works with almost all calls.  We can always pass another variable to that function to further granulize the way the responses are tested.
Let me know if you have any thoughts on this.

Reply:
check_response sounds like a great solution